CMakeLists.txt

```diff
@@ -331,20 +333,60 @@ endif()
 set(TECA_VERSION_DESCR ${tmp} CACHE STRING "TECA version descriptor")
 message(STATUS "TECA version ${TECA_VERSION_DESCR}")
 
-# do we have access to a checkout of TECA_data?
-set(tmp OFF)
-set(TECA_DATA_ROOT "/path/to/TECA_data"
-    CACHE PATH "Path to TECA_data repository")
-if (EXISTS "${TECA_DATA_ROOT}")
-    set(tmp ON)
-    message(STATUS "TECA_data -- available")
-elseif (REQUIRE_TECA_DATA)
-    message(FATAL_ERROR "TECA_data -- required but not found")
+# configure for TECA_data
+set(TECA_DATA_ROOT CACHE PATH "Path to TECA_data repository")
+if (REQUIRE_TECA_DATA)
+    # do we have access to a checkout of TECA_data?
+    if (TECA_DATA_ROOT)
+        if (EXISTS "${TECA_DATA_ROOT}")
+            set(tmp ON)
+            message(STATUS "TECA_data -- available")
+        else()
+            message(FATAL_ERROR "TECA_data -- required but user-set path not found")
+        endif()
+    else()
+        set(tmp ON)
+        ExternalProject_Add(TECA_data
+            SVN_REPOSITORY svn://svn.code.sf.net/p/teca/TECA_data
+            UPDATE_COMMAND svn update
+            CONFIGURE_COMMAND "" BUILD_COMMAND ""
+            INSTALL_COMMAND "" LOG_DOWNLOAD 1)
```

Comment:
how verbose is this? is it disruptive to display in the terminal? if it is not too bad doing so may help a user understand what's going on. I'll leave that call up to you.

Reply:
Yes it is a little disruptive especially to the progress percentage-values that we get at the building process.